File: read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function readPdfs() {
    try {
        const newPdf = fs.readFileSync('C:\\Users\\yeswa\\OneDrive\\Desktop\\IMP\\CV.pdf');
        const newText = await pdf(newPdf);
        console.log("--- NEW CV ---");
        console.log(newText.text);

        const oldPdfFilePath = 'C:\\Users\\yeswa\\OneDrive\\Desktop\\PORTFOLIO\\assets\\Yeswanth_Burle_CV.pdf';
        if (fs.existsSync(oldPdfFilePath)) {
            const oldPdf = fs.readFileSync(oldPdfFilePath);
            const oldText = await pdf(oldPdf);
            console.log("\n\n--- OLD CV ---");
            console.log(oldText.text);
        } else {
            console.log("\n\n--- OLD CV ---");
            console.log("Old CV not found at the expected path.");
        }
    } catch (e) {
        console.error("Error reading PDFs:", e);
    }
}

readPdfs();

File: read_pdf.py
import sys

try:
    import PyPDF2
except ImportError:
    pass

def get_pdf_text(path):
    text = ""
    try:
        with open(path, 'rb') as f:
            reader = PyPDF2.PdfReader(f)
            for page in reader.pages:
                text += page.extract_text() + "\n"
    except Exception as e:
        text = f"Error reading {path}: {e}"
    return text

old_cv = r"C:\Users\yeswa\OneDrive\Desktop\PORTFOLIO\assets\Yeswanth_Burle_CV.pdf"
new_cv = r"C:\Users\yeswa\OneDrive\Desktop\IMP\CV.pdf"

old_text = get_pdf_text(old_cv)
new_text = get_pdf_text(new_cv)

with open('cv_output_old.txt', 'w', encoding='utf-8') as f:
    f.write(old_text)

with open('cv_output_new.txt', 'w', encoding='utf-8') as f:
    f.write(new_text)

print("Wrote output to cv_output_old.txt and cv_output_new.txt")
